### PR TITLE
Add a Pod-based example for NFS volumes

### PR DESCRIPTION
**Pull request by mac-chaffee**
The existing documentation for NFS volumes was missing an example for mounting NFS volumes directly in Pods. The linked example only shows how to mount NFS volumes via PersistentVolumes.

I've added a Pod-based example. 

I've also added a note about how you cannot specify mount options inside a Pod spec (like you can in a PersistentVolume) and listed some alternatives since there is no plan to support a new `mountOptions` setting in Pods: https://github.com/kubernetes/design-proposals-archive/blob/main/storage/mount-options.md#preventing-users-from-specifying-mount-options-in-inline-volume-specs-of-pod

Other relevant issues:
* https://github.com/kubernetes/kubernetes/issues/96284
* https://github.com/kubernetes/kubernetes/pull/107240
* https://github.com/kubernetes/kubernetes/issues/60729
* https://github.com/kubernetes/community/pull/321#issuecomment-279822350

### CODE REVIEW

**Comment by divya-mohan0209** [comment]
@kubernetes/sig-storage-pr-reviews: Please could you check this from a technical standpoint & provide signoff/suggestions?

File: content/en/docs/concepts/storage/volumes.md [hunk 1]
```diff
@@ -762,11 +762,33 @@ unmounted. This means that an NFS volume can be pre-populated with data, and
...
+    nfs:
+      server: my-nfs-server.example.com
+      path: /my-nfs-volume
+      readonly: true
+```
+
 {{< note >}}
 You must have your own NFS server running with the share exported before you can use it.
+
+Also note that you can't specify NFS mount options in a Pod spec. You can either set mount options server-side or use [/etc/nfsmount.conf](https://man7.org/linux/man-pages/man5/nfsmount.conf.5.html).
```
Comment (jsafrane) [inline content/en/docs/concepts/storage/volumes.md:788]: Users can use also PVs, which allow mount options.

**Comment by jsafrane** [comment]
lgtm from sig-storage side

**Comment by sftim** [comment]
/lgtm

for SIG Docs

**Pull request merged**